### DISCUSSION TITLE
Fetch is not defined

### DISCUSSION THREAD

**Discussion by ortizvinicius** [vitest-dev/vitest#1206] [Q&A] [answered]
Hi, i am trying to use vitest in a Vue application where i use fetch, but i get this error:
```
ReferenceError: fetch is not defined
 ❯ Module.login src/composables/auth.js:7:3
      5|
      6| async function login(options, user, pass){
      7|   return await fetch(`${options.server}/auth/login`, {
       |   ^
      8|     method: 'POST',
      9|     headers: {
```

**Comment by sheremet-va** [comment, marked as answer]
Pretty self-explanatory tbh :P
You need to provide `fetch` to the global namespace. Node doesn't have `fetch` builtin (actually, it does, - but in Node 18)

**Reply by jsjoeio** [reply]
But if you configure Vitest to use `js-dom`, shouldn't `fetch` be available? It works with `document`. See example: https://vitest.dev/api/#resolves

**Reply by ortizvinicius** [reply]
@jsjoeio 
I am using jsdom, but had to declare fetch anyway
```
export default defineConfig({
  plugins: [
    Vue(),
  ],
  test: {
    globals: true,
    environment: 'jsdom',
    maxThreads: 1,
    minThreads: 1
  },
})
```

`global.fetch = (url, options) => fetch(cookieJar, url, options);`

**Reply by sheremet-va** [reply]
jsdom doesn't provide `fetch` out of the box.

**Reply by jsjoeio** [reply]
ah! but `happy-dom` does. thanks for clarifying y'all!

**Comment by PankajSinghKerola** [comment]
Just install node 18 fetch is globally available there

**Reply by anztrax** [reply]
thanks @PankajSinghKerola 👍 .. u save my day

**Reply by philipp-serfling** [reply]
~But even with node v18 I am getting the error `ReferenceError: fetch is not defined`. 
Do I need to configure something else?~. 
And make sure to set `environment: "node"` in the config or set nothing at all, as `node` is the default.
https://vitest.dev/config/#environment

**Comment by thinkasany** [comment]
https://nodejs.org/en/blog/announcements/v18-release-announce 

![image](https://github.com/vitest-dev/vitest/assets/117748716/31b12291-2b4e-4e95-bda4-127e07cd38e5)
